src/core/decoder.js

```diff
@@ -139,10 +118,19 @@ function getLineNormal(a, b) {
     return normalize([-dy, dx]);
 }
 
+function getCornerVector(a, b, c) {
+    const u = normalize([a[0] - b[0], a[1] - b[1]]);
+    const v = normalize([c[0] - b[0], c[1] - b[1]]);
+    const sin = - u[1] * v[0] + u[0] * v[1];
+    if (sin !== 0) {
+        return [(u[0] + v[0]) / sin, (u[1] + v[1]) / sin];
+    }
```

Comment:
I think it would be better (and equivalent) to return here the normal instead of relying on the caller for managing the corner-case, which I guess it's that `B` it's in the line `AC`, therefore, the angle between `AB` and `CB` is zero. (The returned value should be the normal of `AC`.

Reply:
`na` and `nb` are initialized with lineNormal. The only condition to be replaced by the jointNormal is when the jointNormal exists (when sin!=0).
I think returning the lineNormal inside the jointNormal function is mixing things because in the singularity case the jointNormal does not exists so it must return a null value.